docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -165,79 +167,66 @@ on the same platform.
 With quote-ex, quote generation is always done out-of-process
 by working with an AESM service on the local platform.
 
-An SGX platform can have either the quote-ex library or the DCAP library,
+### Priority between the SGX DCAP and quote-ex Libraries
+
+An SGX platform can have either the DCAP library or the quote-ex library,
 or both of them installed.
-Installation and configuration of the AESM background service is independent
-of the installation of the two libraries.
 
-When both libraries and the background services are all installed and configured,
-the quote-ex should library takes precedence, as it supports more evidence
-formats.
+* When only the DCAP library is installed, it will be used.
+* When only the quote-ex library is installed, it will be used.
```

Comment:
Thanks for laying out the logic here, it's useful. I just noticed that this isn't quite how the implementation in #3268 works though: by checking for the regkey setting first; if `SGX_AESM_ADDR` is not set, it defaults to DCAP, even if it only quote-ex is installed. 

I think the design here is fine, but the implementation needs to handle this extra bit of logic (use quote-ex if regkey not set and dcap is absent).

Reply:
@CodeMonkeyLeet, I think the description between line 134 and 183 states the implementation.

Reply:
After viewing the implementation in `host\sgx\sgxquote.c` line 175, I still see that the implementation is inline with the described logic. Before checking `SGX_AESM_ADDR`, the code tries to load the DCAP library. If the DCAP is missing, the code will not check `SGX_AESM_ADDR` and try quote-ex directly.

Let me know if I still missed something.

Reply:
@shnwc No, you're right, the existing code does take care of it. I was confused by the fact that we have a redundant `_load_sgx_dcap_ql` call in the else clause for `_use_quote_ex_library` conditional checks (effectively, `_use_quote_ex_library` would have loaded one or the other for the lifetime of the process by the time it returns.)

Side note: Consider reversing the order of the conditional checks in `_load_quote_ex_library_once` so that the DCAP load is elided if it's going to fall through to quote_ex load anyway you don't end up with an unused library load. 
```c
if (SGX_DCAP_IN_PROCESS_QUOTING() && _load_sgx_dcap_ql())
```

Reply:
Good point! Will make the change and create a new minor PR for it.